File: problems/3375-minimum-operations-to-make-array-values-equal-to-k/solution.py
# Link: https://leetcode.cn/problems/minimum-operations-to-make-array-values-equal-to-k/
# Level: Easy
# Date: 2025-04-09

class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        if min(nums) < k:
            return -1
        s = set()
        for num in nums:
            if num > k:
                s.add(num)
        return len(s)

File: problems/3375-minimum-operations-to-make-array-values-equal-to-k/solution.cpp
// Link: https://leetcode.cn/problems/minimum-operations-to-make-array-values-equal-to-k/
// Level: Easy
// Date: 2025-04-09

class Solution {
    public:
        int minOperations(vector<int>& nums, int k) {
            int n = nums.size();
            int res = -1;
            unordered_set<int> s;
            for (auto num : nums) {
                if (num < k) {
                    return -1;
                } else if (num > k) {
                    s.insert(num);
                }
            }
            res = s.size();
            return res;
        }
};